src/Env.ml

```diff
@@ -20,6 +21,8 @@ module Make (Key : sig
   include Equal.S with type t := t
 
   val context_kind : string
+
+  val to_string : t -> string
```

Comment:
We should probably reuse the `Stringable` interface of the Base library. Kind of like this:
```
  include Stringable.S with type t := t
```

Reply:
I think `Stringable` looks like an interface for serializable objects since it contains exact two methods `to_string : t -> string` and `of_string : string -> t`. We can replace `Id.to_string` and `Id.mk` with `Stringable`, but it's not the serialization (e. g. `mk` can change its type).   

Something like Haskell's `Show` with single method `show : t -> string` is more appropriate here. 
Maybe pxx plugins https://github.com/ocaml-ppx/ppx_deriving#plugin-show.

Reply:
In Base, there is also a special module for identifiers: https://github.com/janestreet/base/blob/master/src/identifiable.mli